central/group/datastore/datastore_impl.go

```diff
@@ -231,11 +231,9 @@ func (ds *dataStoreImpl) RemoveAllWithAuthProviderID(ctx context.Context, authPr
 
 func (ds *dataStoreImpl) RemoveAllWithEmptyProperties(ctx context.Context) error {
 	// Search through all groups and verify whether any group exists with empty properties and attempt to delete them.
-	isEmptyGroupPropertiesF := func(props *storage.GroupProperties) bool {
-		if props.GetAuthProviderId() == "" && props.GetKey() == "" && props.GetValue() == "" {
-			return true
-		}
-		return false
+	isEmptyGroupPropertiesF := func(group *storage.Group) bool {
```

Comment:
The code behind it is quite old and we were unsure whether we even wanted to reintroduce this. So, if anything, I'd rather like to remove it altogether instead of adding another constraint (sadly, the comments on the original PR were quite sparse _why_ this was required to begin with..).